File: frontend/src/types/interview.ts
export interface Interview {
  id: number
  applicationId: number
  interviewTime: string
  location?: string
  interviewer?: string
  result?: InterviewResult
  feedback?: string
  createdAt?: string
  updatedAt?: string
}

export enum InterviewResult {
  PENDING = 'PENDING',
  PASSED = 'PASSED',
  FAILED = 'FAILED'
}

export interface InterviewDTO {
  id?: number
  applicationId: number
  interviewTime: string
  location: string
  interviewer?: string
  result?: string
  feedback?: string
}

File: backend/src/main/java/com/example/backend/entity/Interview.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "interviews")
public class Interview {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "application_id", nullable = false)
    private Long applicationId;

    @Column(name = "interview_time", nullable = false)
    private LocalDateTime interviewTime;

    @Column(length = 200)
    private String location;

    @Column(length = 50)
    private String interviewer;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private Result result;

    @Column(columnDefinition = "TEXT")
    private String feedback;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum Result {
        PENDING, PASSED, FAILED
    }
}
